spark-benchmarks/src/main/scala/AdvertisingSpark.scala

```diff
@@ -91,20 +91,18 @@ object KafkaRedisAdvertisingStream {
     //each record in the RDD: key:(campaign_id : String, window_time: Long),  Value: (ad_id : String)
     //DStream[((String,Long),String)]
 
-    val groupedByCampaignTime = campaign_timeStamp.groupByKey()
-    // DStream[((String,Long), Iterable[String])]
+    // since we're just counting use reduceByKey
+    val totalEventsPerCampaignTime = campaign_timeStamp.mapValues(_ => 1).reduceByKey(_ + _)
```

Comment:
I know the Spark Api, but:

campaign_timeStamp.mapValues(_ => 1).reduceByKey(_ + _)

You can change this two transformations for only one combineByKey